app/global/MainModule.java

```diff
@@ -207,6 +213,20 @@ protected void configure() {
                 .asEagerSingleton();
     }
 
+    @Singleton
+    @Provides
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Guice
```

Comment:
disclaimer: I'm unsure what the behavior here will be

would this result in "spiky" cpu usage? that is to say, if we ran every 30 minutes, would it look something like the following timeline?
```
[alert schedule starts] [5 minutes of CPU at 100%] [25 minutes of CPU at ~0%]
```

and, if so, I don't even know if that's a problem

but what i'm getting at is, maybe we want to introduce some like noise/stutter and spread this out?

again, to reiterate: I Have No Idea If This Is A Thing To Be Concerned About (tm) (copyright)

Reply:
I definitely expect some sort of spikiness, we already see something like this with rollups when the hour/day rolls over. I don't think this will necessarily be an issue out the gate, but @vjkoskela and I had discussed this as another motivating factor for query caching which I believe is already on our roadmap.

As for any noise/jittering - I think I'd like to see this in the [JobCoordinator] instead of dealing with it at the per-job level. At the least we may want to introduce some form of queueing since I think right now we just spawn an actor for [every possible job]. This would broaden any spikes not by messing with scheduling but by limiting the number of concurrently executing jobs. For alerts this should already be handled by limits on [the query executor], but it would still be nice to have this guarantee more generally.

[JobCoordinator]: https://github.com/ArpNetworking/metrics-portal/blob/7b51cca93cfd1d9ebc0acba83b2a76942aa37862/app/com/arpnetworking/metrics/portal/scheduling/JobCoordinator.java
[the query executor]: https://github.com/ArpNetworking/metrics-portal/blob/e1466b6fb4ef5602caa25842248cca9793925fb6/app/global/MainModule.java#L353
[every possible job]: https://github.com/ArpNetworking/metrics-portal/blob/04921f3c7e1980759815e04c90bd9d0e3a3c0b34/app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java#L66